src/debuginfo.cpp

```diff
@@ -390,7 +399,7 @@ void JITDebugInfoRegistry::registerJITObject(const object::ObjectFile &Object,
 
 void jl_register_jit_object(const object::ObjectFile &Object,
                             std::function<uint64_t(const StringRef &)> getLoadAddress,
-                            std::function<void *(void *)> lookupWriteAddress) JL_NOTSAFEPOINT
+                            std::function<void *(void *)> lookupWriteAddress)
```

Comment:
I am pretty sure you are forbidden from removing this annotation. There were locks that needed to be held in the caller, so you are now violating the lock hierarchy.

Reply:
(GCChecker probably doesn't yet know how to model `std::lock_guard`, so it didn't automatically detect this violation)

Reply:
Is holding that lock required while running this function? I see two call sites, only one of which is holding the lock.

Reply:
I believe OrcJIT is holding other locks that can't be seen. Why aren't you doing this as part of codegen, when transferring it to the inflight list?